src/extensions/default/JavaScriptCodeHints/HintUtils.js

```diff
@@ -145,7 +145,7 @@ define(function (require, exports, module) {
             t.kind = kind;
             t.origin = "ecmascript";
             if (kind === "string") {
-                if (/[\\\\]*[^\\]"/.test(t.value)) {
+                if (/[^\\]"/.test(t.value)) {
```

Comment:
Good stuff @xiemaisi, merging this in and will land in 1.11. Hopefully we get the other fixes found by `lgtm` in too.